schema/auth.json

```diff
@@ -0,0 +1,55 @@
+{
+  "$id": "https://serverlessworkflow.io/schemas/0.7/auth.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "Serverless Workflow specification - auth schema",
+  "type": "object",
+  "auth": {
+    "oneOf": [
+      {
+        "type": "string",
+        "format": "uri",
+        "description": "URI to a resource containing auth definitions (json or yaml)"
+      },
+      {
+        "type": "array",
+        "description": "Workflow auth definitions",
+        "items": {
+          "type": "object",
+          "$ref": "#/definitions/authdef"
+        },
+        "additionalItems": false,
+        "minItems": 1
+      }
+    ]
+  },
+  "required": [
+    "auth"
+  ],
+  "definitions": {
+    "authdef": {
```

Comment:
I think we need an `anyOf` here, with the following schemas:

## Basic
```
{
  "name": "MyBasicAuth",
  "scheme": "basic",
  "properties": {
    "username": "MyUserName",
    "password":  "MyPassword"
  }
}
```

## Bearer

```
{
  "name": "MyBearerAuth",
  "scheme": "bearer",
  "properties": {
    "token": "MyToken"
  }
}
```

## OpenID Connect

```
{
  "name": "MyBearerAuth",
  "scheme": "oidc",
  "properties": {
    "grantType": "resourceOwner | clientCredentials | tokenExchange",
    "clientId": "MyClientId", #required
    "clientSecret": "MyClientSecret", #optional
    "username": "MyUserName", #only for resourceOwner
    "password": "MyPassword" #only for resourceOwner,
    "scope": "openid otherScopes"
  }
}
```

Reply:
added

Reply:
only difference is that instead of "properties" i called it "info"